lib/SILOptimizer/Transforms/PerformanceInliner.cpp

```diff
@@ -362,6 +362,55 @@ bool SILPerformanceInliner::isProfitableToInline(FullApplySite AI,
                         NumCallerBlocks / BlockLimitDenominator;
   Benefit -= blockMinus;
 
+  // If we have profile info - use it for final inlining decision.
+  auto *bb = AI.getInstruction()->getParent();
+  auto bbIt = BBToWeightMap.find(bb);
+  if (bbIt != BBToWeightMap.end()) {
```

Comment:
This is already a monster function. Can you extract this code into a separate function?